### ISSUE TITLE
Loading savestate with different resolution causes crash

### ISSUE THREAD

**Issue by slamotas** [clementgallet/libTAS#103]
This crash occurred when playing Terminal Velocity in DOSBox. If SVGA mode is turned on in the graphics options, the resolution of DOSBox increases once the first stage starts. If a state from the menu is loaded after the resolution change, DOSBox crashes.

**Comment by Matte27** [comment]
Seems to work fine with meatboy, although the game window isn't resized properly.

**Comment by slamotas** [comment]
Yeah, in PCem it will load the state without crashing, but not resize the window back. 
I never posted any error message, here's what I get with gdb for this crash:

`X Error of failed request:  BadShmSeg (invalid shared segment parameter)
  Major opcode of failed request:  130 (MIT-SHM)
  Minor opcode of failed request:  3 (X_ShmPutImage)
  Segment id in failed request:  0x2c00015
  Serial number of failed request:  1498
  Current serial number in output stream:  1499
 [Thread 0x7fffec45c700 (LWP 30582) exited]
 [Inferior 1 (process 30559) exited with code 01]
`

This is not a huge priority, I can work around this and there aren't really any frame perfect inputs that have to be done between resolution changes.
